packages/clerk-js/src/core/clerk.ts

```diff
@@ -975,7 +975,12 @@ export default class Clerk implements ClerkInterface {
     return false;
   };
 
-  #shouldSyncWithPrimary = () => this.#options.isSatellite && this.#handleSyncedQueryParam();
+  #shouldSyncWithPrimary = () => {
+    if (typeof this.#options.isSatellite === 'function') {
```

Comment:
🙃  It's more verbose but easier to follow the flow as the if clause handles the isSatellite calculation and then based on that we invoke another action.

```suggestion
let isSatellite = false;
if (typeof this.#options.isSatellite === 'function') {
  isSatellite = this.#options.isSatellite(new URL(window.location.href));
} else {
  isSatellite = !!this.#options.isSatellite;
}

if (isSatellite) {
  this.#handleSyncedQueryParam();
}
```

Reply:
@SokratisVidros  How about moving the above lines 979-984 inside a public getter ?

Reply:
Fine by me